### PR TITLE
Remove Vault Connection from the manager.

### PR DESCRIPTION
**Pull request by revit13**
This PR removes Vault connection from the manager as it is currently not used in the manager code.

### CODE REVIEW

File: manager/controllers/utils/configuration.go [hunk 1]
```diff
@@ -15,9 +15,16 @@ import (
...
-	VaultModulesRole                  string = "VAULT_MODULES_ROLE"
+	// FIXME: Current communication between the manager and Vault assumes a Vault
+	// token which is passed as an env var.
+	// This should be changed to the following:
+	// 1. get a Vault token form Vault using the manager role
+	// 2. continue communication with Vault using this Vault token
+	// The manager role should be passed as env var after it was configured
+	// with the appropriate permissions for the manager.
+	VaultSecretKey   string = "VAULT_TOKEN"
+	VaultTTLKey      string = "VAULT_TTL"
```
Comment (shlomitk1) [inline manager/controllers/utils/configuration.go:26]: We are not using neither VAULT_TOKEN nor VAULT_TTL, so why keep it?

Comment (shlomitk1) [reply manager/controllers/utils/configuration.go:26]: Not sure what is the purpose of this PR. 
1. manager role env is not introduced, only mentioned in a comment
2. If vault token should be obtained by the manager, why is it still in the configmap and configuration.go file?

**Comment by revit13** [comment]
Following a private talk with @shlomitk1 I changed the pr to remove Vault connection from the manager code and updated the pr description accordently.

**Review by shlomitk1 — APPROVED** [review]

**Comment by shlomitk1** [comment]
@revit13 Please remove VAULT_TTL and VAULT_TOKEN from the chart values/ m4d-config as well.

**Comment by roee88** [comment]
@revit13 is there an issue for changing to make the manager setup policies dynamically?

**Comment by revit13** [comment]
@shlomitk1 Thanks will change that. @roee88 in issue  #585 the last task refers to that.

**Comment by revit13** [comment]
@shlomitk1 I left charts/m4d/templates/vault-credentials.yaml as it is still used...

**Comment by shlomitk1** [comment]
> 
> 
> @shlomitk1 I left charts/m4d/templates/vault-credentials.yaml as it is still used...

Secret should be left, but the reference to it in the manager deployment is no longer required.
https://github.com/mesh-for-data/mesh-for-data/blob/f06f70e3c6a3dcf643d7416a8578fba4f14a50ae/charts/m4d/templates/manager-deployment.yaml#L81-L84

**Comment by revit13** [comment]
> 
> 
> > @shlomitk1 I left charts/m4d/templates/vault-credentials.yaml as it is still used...
> 
> Secret should be left, but the reference to it in the manager deployment is no longer required.
> 
> https://github.com/mesh-for-data/mesh-for-data/blob/f06f70e3c6a3dcf643d7416a8578fba4f14a50ae/charts/m4d/templates/manager-deployment.yaml#L81-L84

Thanks @shlomitk1 Fixed.

File: pkg/vault/vault_interface_test.go [hunk 2]
```diff
@@ -22,10 +22,11 @@ func Log(t *testing.T, label string, err error) {
 
 func TestCredentialManagerInterface(t *testing.T) {
 	var err error
+	vaultToken := "dummyToken"
 	// TODO add environment variables to test against a real vault instance
 	os.Setenv("RUN_WITHOUT_VAULT", "1")
-	t.Logf("Token = " + utils.GetVaultToken())
-	conn, err := InitConnection(utils.GetVaultAddress(), utils.GetVaultToken())
+	t.Logf("Token = " + vaultToken)
```
Comment (roee88) [inline pkg/vault/vault_interface_test.go:28]: No tokens in logs

**Review by roee88 — CHANGES_REQUESTED** [review]

File: pkg/vault/vault_interface_test.go (hunk 2 above)
Comment (revit13) [reply pkg/vault/vault_interface_test.go]: Fixed Thanks

**Review by roee88 — APPROVED** [review]

**Pull request merged**